internal/proto/opamp.proto

```diff
@@ -410,30 +408,39 @@ message AgentPackageAvailable {
 // Status reporting
 
 message AgentDescription {
-    // Reverse FQDN that uniquely identifies the agent type, e.g. "io.opentelemetry.collector"
-    string agent_type = 1;
-
-    // Version number of the agent build. The server can use this information to
-    // decide if it wants to offer a package of a different version to the agent
-    // via PackageAvailable message.
-    string agent_version = 2;
-
-    // Attributes that describe the agent and the environment it runs in.
+    // Attributes that identify the agent.
     // Keys/values are according to OpenTelemetry semantic conventions, see:
     // https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions
     //
+    // For standalone running Agents (such as OpenTelemetry Collector) the following
+    // attributes SHOULD be specified:
+    // - service.name should be set to a reverse FQDN that uniquely identifies the
+    //   agent type, e.g. "io.opentelemetry.collector"
+    // - service.namespace if it is used in the environment where the Agent runs.
+    // - service.version should be set to version number of the Agent build.
+    // - service.instance.id should be set. It may be be set equal to the Agent's
+    //   instance uid (equal to ServerToAgent.instance_uid field) or any other value
+    //   that uniquely identifies the Agent in combination with other attributes.
+    // - any other attributes that are necessary for uniquely identifying the Agent's
+    //   own telemetry.
+    //
+    // The Agent SHOULD also include these attributes in the Resource of its own
+    // telemetry. The combination of identifying attributes SHOULD be sufficient to
+    // uniquely identify the Agent's own telemetry in the destination system to which
+    // the Agent sends its own telemetry.
+    repeated KeyValue identifying_attributes = 1;
```

Comment:
I went with key/value pair lists, similar to OTLP. We could also do a map[string]KeyValue, but we likely have to have to require uniqueness across identifyin and non-identifying attributes, so the map is not much of an improvement over a list (and is slower).